sdk/storage/Azure.Storage.Files.Shares/perf/Azure.Storage.Files.Shares.Perf/Scenarios/DownloadFile.cs

```diff
@@ -0,0 +1,104 @@
+﻿//Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Test.Perf;
+
+namespace Azure.Storage.Files.Shares.Perf.Scenarios
+{
+    /// <summary>
+    /// The performance test scenario focused on downloading files from the Azure Files Shares storage.
+    /// </summary>
+    /// <seealso cref="Azure.Test.Perf.PerfTest{SizeOptions}" />
+    public sealed class DownloadFile : PerfTest<SizeOptions>
+    {
+        /// <summary>
+        /// The client to interact with an Azure storage share.
+        /// </summary>
+        private ShareClient _shareClient;
+
+        /// <summary>
+        /// The client to interact with an Azure storage file inside an Azure storage share.
+        /// </summary>
+        private ShareFileClient _fileClient;
+
+        /// <summary>
+        /// Local stream uploaded to Azure storage.
+        /// </summary>
+        private readonly Stream _stream;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DownloadFile"/> class.
+        /// </summary>
+        /// <param name="options">The set of options to consider for configuring the scenario.</param>
+        public DownloadFile(SizeOptions options) : base(options)
+        {
+            _stream = RandomStream.Create(options.Size);
+        }
+
+        public override void Dispose(bool disposing)
+        {
+            _stream.Dispose();
+            base.Dispose(disposing);
+        }
+
+        public override async Task SetupAsync()
+        {
+            await base.SetupAsync();
+
+            // See https://docs.microsoft.com/en-us/rest/api/storageservices/naming-and-referencing-shares--directories--files--and-metadata for
+            // restrictions on file share naming.
+            _shareClient = new ShareClient(PerfTestEnvironment.Instance.FileSharesConnectionString, Guid.NewGuid().ToString());
```

Comment:
I think some of this code is copied across multiple tests, so I would put it into a common base class.